pkg/authenticator/target_cluster_client.go

```diff
@@ -219,3 +222,45 @@ func (tcc *targetClusterClient) CheckNamespace(ctx context.Context, namespace st
 
 	return true
 }
+
+// CreateSecret on the workload cluster
+//
+// It must only be called with an initialized target cluster client.
+func (tcc *targetClusterClient) CreateSecret(ctx context.Context, secret *corev1.Secret) error {
+	if tcc.clientConfig == nil {
+		tcc.logger.Error(fmt.Errorf("client is not initialized"), "creating secret")
+		return fmt.Errorf("client is not initialized")
+	}
+
+	restConfig, err := tcc.ToRESTConfig()
+	if err != nil {
+		tcc.logger.V(6).Error(err, "creating rest config")
+		return err
+	}
+
+	k8sClient, err := client.New(restConfig, client.Options{})
```

Comment:
We really need to fix that targetClusterClient so it handles creating the correct client(s). Out of scope here.